### PR TITLE
MAT-5903: Delete Attribute Chip

### PR DESCRIPTION
**Pull request by ethankaplan**
## MADiE PR

Jira Ticket: [MAT-5903](https://jira.cms.gov/browse/MAT-5903)
(Optional) Related Tickets:

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included
* [ ] No extemporaneous files are included (i.e Complied files or testing results)
* [ ] This PR is into the **correct branch**.
* [ ] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [ ] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

### CODE REVIEW

File: src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DataElementsCard/DataElementsCard.test.tsx [hunk 1]
```diff
@@ -903,6 +910,24 @@ export const testDataElements = [
     description: "Device, Order: Cardiopulmonary Arrest",
     negationRationale: true,
   },
+  {
+    //15
```
Comment (jkotanchik-SB) [inline src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DataElementsCard/DataElementsCard.test.tsx]: 15?

Comment (jkotanchik-SB) [reply src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DataElementsCard/DataElementsCard.test.tsx]: Ah, it's the 15th item in the array.

File: src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DataElementsCard/AttributeChip.tsx [hunk 2]
```diff
@@ -2,13 +2,30 @@ import React from "react";
 import { IconButton } from "@mui/material";
 import CancelIcon from "@mui/icons-material/Cancel";
 import "./AttributeChipList.scss";
-const AttributeChip = (AttributeChipProps: { text }) => {
-  const { text } = AttributeChipProps;
+
+interface AttributeChipProps {
+  text: string;
+  onDeleteAttributeChip?: (deletedChip) => void;
+  index?: string;
```
Comment (jkotanchik-SB) [inline src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DataElementsCard/AttributeChip.tsx]: Should these be required fields? Will bad things happen if they aren't passed?

File: src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DataElementsCard/DataElementsCard.test.tsx [hunk 3]
```diff
@@ -997,6 +1021,36 @@ describe("DataElementsCard", () => {
...
+      renderDataElementsCard(
+        "attributes",
+        jest.fn,
+        testDataElements[15],
+        jest.fn
+      )
+    );
+
+    expect(await screen.findByText("Result: 1")).toBeInTheDocument();
+  });
```
Comment (adongare) [inline src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DataElementsCard/DataElementsCard.test.tsx]: seems a redundant test case. can be removed?

File: src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DataElementsCard/DataElementsCard.test.tsx [hunk 4]
```diff
@@ -997,6 +1021,36 @@ describe("DataElementsCard", () => {
...
+    const resultChip = await screen.findByText("Result: 1");
+    expect(resultChip).toBeInTheDocument();
+    const closeButton = await screen.findByTestId("delete-chip-button-0");
+    expect(closeButton).toBeInTheDocument();
+    userEvent.click(closeButton);
+    expect(closeButton).not.toBeInTheDocument();
+    expect(resultChip).not.toBeInTheDocument();
+  });
+
   it("Should add new Codes", async () => {
```
Comment (adongare) [inline src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DataElementsCard/DataElementsCard.test.tsx:1041]: `test` and `it` both are the same but good to be consistent and follow one pattern.

**Review by adongare — APPROVED** [review]

**Pull request merged**